pennylane_lightning_gpu/src/algorithms/AdjointDiffGPU.hpp

```diff
@@ -586,36 +852,33 @@ template <class T = double> class AdjointJacobianGPU {
             const auto last = static_cast<std::size_t>(
                 std::ceil((obs.size() * (i + 1) / num_chunks) - 1));
 
-            std::promise<std::vector<std::vector<T>>> jac_subset_promise;
+            std::promise<std::vector<T>> jac_subset_promise;
             futures.emplace_back(jac_subset_promise.get_future());
 
-            auto adj_lambda =
-                [&](std::promise<std::vector<std::vector<T>>> j_promise) {
-                    // Ensure No OpenMP threads spawned;
-                    // to be resolved with streams in future releases
-                    omp_set_num_threads(1);
-                    // Grab a GPU index, and set a device tag
-                    const auto id = dp.acquireDevice();
-                    DevTag<int> dt_local(id, 0);
-                    dt_local.refresh();
-
-                    // Create a sv copy on this thread and device; may not be
-                    // necessary, could do in adjoint calculation directly
-                    StateVectorCudaManaged<T> local_sv(ref_data, length,
-                                                       dt_local);
-
-                    // Create local store for Jacobian subset
-                    std::vector<std::vector<T>> jac_local(
-                        (last - first + 1),
-                        std::vector<T>(trainableParams.size(), 0));
-                    adjointJacobian(
-                        local_sv.getData(), length, jac_local,
-                        {obs.begin() + first, obs.begin() + last + 1}, ops,
-                        trainableParams, apply_operations, dt_local);
-
-                    j_promise.set_value(std::move(jac_local));
-                    dp.releaseDevice(id);
-                };
+            auto adj_lambda = [&](std::promise<std::vector<T>> j_promise) {
+                // Ensure No OpenMP threads spawned;
+                // to be resolved with streams in future releases
+                omp_set_num_threads(1);
+                // Grab a GPU index, and set a device tag
+                const auto id = dp.acquireDevice();
+                DevTag<int> dt_local(id, 0);
+                dt_local.refresh();
+
+                // Create a sv copy on this thread and device; may not be
+                // necessary, could do in adjoint calculation directly
+                StateVectorCudaManaged<T> local_sv(ref_data, length, dt_local);
+
+                // Create local store for Jacobian subset
+                std::vector<T> jac_local(
+                    (last - first + 1) * trainableParams.size(), 0);
+                adjointJacobian(local_sv.getData(), length, jac_local,
+                                {obs.begin() + first, obs.begin() + last + 1},
+                                ops, trainableParams, apply_operations,
+                                dt_local);
+
+                j_promise.set_value(std::move(jac_local));
+                dp.releaseDevice(id);
+            };
```

Comment:
Does the concatenation functionality below this lambda need to be updated to reflect the dimensionality change of the Jacobian data-sets??